connect_deprish_to_eia.py

```diff
@@ -111,9 +111,20 @@ def prep_deprish(file_path_deprish, plant_parts_df,
     return deprish_ids
 
 
-def prep_master_parts_eia(plant_parts_df, key_mul):
+def prep_master_parts_eia(plant_parts_df, deprish_df, key_mul):
     """Prepare the EIA master plant parts."""
-    plant_parts_df[key_mul] = pudl.helpers.cleanstrings_series(
+    # restrict the possible matches to only those that match on the
+    # RESTRICT_MATCH_COLS
+    plant_id_pudls_d = deprish_df.loc[:, 'plant_id_pudl'].unique().tolist()
+    report_years_d = deprish_df.loc[:, 'report_year'].unique().tolist()
+    utility_ids_d = deprish_df.loc[:, 'utility_id_pudl'].unique().tolist()
+
+    plant_parts_df = plant_parts_df.loc[
+        (plant_parts_df.plant_id_pudl.isin(plant_id_pudls_d))
+        & (plant_parts_df.report_year.isin(report_years_d))
+        & (plant_parts_df.utility_id_pudl.isin(utility_ids_d))]
+
```

Comment:
I was playing with ways to make this process actually use `RESTRICT_MATCH_COLS` but they all felt contrived and harder to see what was happening. But if you have any ideas lmk.

Reply:
thanks for the tip on this! It looks like this works.

```
options = list(deprish_df[RESTRICT_MATCH_COLS].drop_duplicates()
               .itertuples(index=False))
plant_parts_df = (plant_parts_df.set_index(RESTRICT_MATCH_COLS)
                  .loc[options].reset_index())